### ISSUE TITLE
Consider removing `ModuleParams`

### ISSUE THREAD

**Issue by fvictorio** [nomiclabs/hardhat-ignition#56]
I might be missing something here, but I think we could do without the whole `ModuleParams` thing.

If I'm understanding correctly, the idea is to be able to (somehow) configure the `ModuleParams` to be, for example, an object like `{ decimals : 18 }` and then they can be used inside modules to do things like:

```js
m.contract("Token", "MyToken", m.decimals)
```

But since the modules are code, I don't see why users couldn't just have a `constants.ts` file (or whatever) and use that. I also don't quite like the idea of adding extra properties to the `ModuleBuilder` class as a mechanism to share data.

**Comment by alcuadrado** [comment]
I didn't understand this initially, but I think it could play really well with the hardhat plugin.

Imagine having a config like this:

```
networks: {
   mainnet: {
      ignitionParams : {...}
   },
   hardhat: {
      ignitionParams : {...}
   }
}
```

But now that I write this down, we should consider how this would work when importing third-party modules.

**Comment by fvictorio** [comment]
Yeah, that use case makes sense to me. I still think it shouldn't be part of the `ModuleBuilder` class though.

**Comment by fvictorio** [comment]
One idea that should cover a lot of scenarios: expose the `chainId` of the network to modules. Or maybe the whole provider?
